### PR TITLE
added support for OR and NOT queries

### PR DESCRIPTION
**Pull request by StigNorland**

### CODE REVIEW

File: search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java [hunk 1]
```diff
@@ -122,9 +120,7 @@ private QueryBuilderSourceWrapper populateSearchRequest(QueryBuilderWrapper quer
 
         builder.size(query.getValue(SIZE).as());
         builder.from(query.getValue(FROM).as());
-        if (query.isPresent(SORT)) {
-            getSortStream(query).forEach(orderTuple -> builder.sort(orderTuple.v1(), orderTuple.v2()));
-        }
+        getSortStream(query).forEach(orderTuple -> builder.sort(orderTuple.v1(), orderTuple.v2()));
```
Comment (StigNorland) [inline search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java:123]: Should handle empty SORT

File: search-commons/src/test/java/no/unit/nva/search2/ResourceQueryTest.java [hunk 2]
```diff
@@ -143,6 +143,10 @@ static Stream<URI> uriProvider() {
             URI.create("https://example.com/"),
             URI.create("https://example.com/?fields=category,title,created_date"),
             URI.create("https://example.com/?query=Muhammad+Yahya&fields=CONTRIBUTOR"),
+            URI.create("https://example.com/?CONTRIBUTOR=https://api.dev.nva.aws.unit.no/cristin/person/1136254"),
+            URI.create("https://example.com/?CONTRIBUTOR_SHOULD="
+                       + "https://api.dev.nva.aws.unit.no/cristin/person/1136254+"
```
Comment (torarnet) [inline search-commons/src/test/java/no/unit/nva/search2/ResourceQueryTest.java:148]: are you using `+ `as a delimiter for multivalued parameters? Should I use it as well for facets? We should discuss

File: search-commons/src/main/java/no/unit/nva/search2/model/ResourceParameterKey.java [hunk 3]
```diff
@@ -46,40 +50,75 @@ public enum ResourceParameterKey implements ParameterKey {
...
+    FUNDING(STRING, MUST, "fundings.identifier|fundings.source.identifier",
             null, PATTERN_IS_FUNDING, null),
     FUNDING_SOURCE(STRING, "fundings.source.identifier|fundings.source.labels"),
+    FUNDING_SOURCE_NOT(STRING, MUST_NOT, "fundings.source.identifier|fundings.source.labels"),
+    FUNDING_SOURCE_SHOULD(STRING, SHOULD, "fundings.source.identifier|fundings.source.labels"),
     ID(STRING, "identifier"),
-    INSTITUTION(STRING, "entityDescription.contributors.affiliation"
-                        + "|entityDescription.contributors.affiliation.name"),
+    ID_NOT(STRING, MUST_NOT, "identifier"),
+    ID_SHOULD(STRING, MUST_NOT, "identifier"),
```
Comment (torarnet) [inline search-commons/src/main/java/no/unit/nva/search2/model/ResourceParameterKey.java:74]: Why does `ID_SHOULD `have `MUST_NOT `as parameter ?

File: search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java [hunk 4]
```diff
@@ -171,36 +167,44 @@ private MultiMatchQueryBuilder multiMatchQuery(ResourceAwsQuery query) {
...
     }
 
+    private QueryBuilder buildQuery(ResourceParameterKey key, String value) {
+        final var searchFields = key.searchFields().toArray(String[]::new);
+        if (hasMultipleFields()) {
+            return QueryBuilders
+                       .multiMatchQuery(escapeSearchString(value), searchFields)
+                       .operator(Operator.AND);
+        }
+            var searchField = searchFields[0];
```
Comment (torarnet) [inline search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java:199]: is indentation correct on these lines?

**Review by torarnet — COMMENTED** [review]
I think you have a wrong mapping in one of the enums, otherwise ok

File: search-commons/src/test/java/no/unit/nva/search2/ResourceQueryTest.java (hunk 2 above)
Comment (StigNorland) [reply search-commons/src/test/java/no/unit/nva/search2/ResourceQueryTest.java:148]: actually it is an encoded space.

File: search-commons/src/main/java/no/unit/nva/search2/model/ResourceParameterKey.java (hunk 3 above)
Comment (StigNorland) [reply search-commons/src/main/java/no/unit/nva/search2/model/ResourceParameterKey.java]: fixed

File: search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java (hunk 4 above)
Comment (StigNorland) [reply search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java]: fixed

File: search-commons/src/main/java/no/unit/nva/search2/model/ResourceParameterKey.java [hunk 5]
```diff
@@ -45,41 +46,76 @@
...
-    TITLE(STRING, "entityDescription.mainTitle", 2F),
-    UNIT(STRING, "entityDescription.contributors.affiliation.id"),
-    USER(STRING, "resourceOwner.owner"),
-    PUBLICATION_YEAR(NUMBER, EQUALS, "entityDescription.publicationDate.year",
+    FUNDING_SOURCE(STRING, Constants.FUNDINGS_SOURCE_IDENTIFIER_FUNDINGS_SOURCE_LABELS),
+    FUNDING_SOURCE_NOT(STRING, MUST_NOT, Constants.FUNDINGS_SOURCE_IDENTIFIER_FUNDINGS_SOURCE_LABELS),
+    FUNDING_SOURCE_SHOULD(STRING, SHOULD, Constants.FUNDINGS_SOURCE_IDENTIFIER_FUNDINGS_SOURCE_LABELS),
+    ID(STRING, Constants.IDENTIFIER),
+    ID_NOT(STRING, MUST_NOT, Constants.IDENTIFIER),
+    ID_SHOULD(STRING, MUST_NOT, Constants.IDENTIFIER),
```
Comment (torarnet) [inline search-commons/src/main/java/no/unit/nva/search2/model/ResourceParameterKey.java]: ID_SHOULD is mapped to MUST_NOT

**Review by torarnet — APPROVED** [review]

**Pull request merged**